scripts/sandboxGenerator.sh

```diff
@@ -0,0 +1,182 @@
+#! /bin/bash
+
+# Script that executes the following operations :
+# 1- Execute the repackage goal to generate the MAC springBoot jar in the state of the repository
+# 2- Executes a maven goal that builds a customized (no real time) Sandbox springboot jar for a given version of Activepivot
+# 3- Run both apps
+# 4- Generate the memory statistics file for the Sandbox
+# 5- Load the generated statisitics in the MAC app
+# 6- Run queries on the MAC app
+# 7- Kill both apps
+
+# REQUIREMENTS :
+# Java 11+
+# Maven 3+
+# ActiveViam artifacts access
+# Valid ActivePivot License
+# jq
+
+# INPUT : 
+# - $1 String : AP version
+
+# OUTPUT : 
+# - File : Query & server logs
+
+# 0- Log execution path & script location path
+
+# Define functions to avoid command duplication
+# Make sure the variables used are defined before calling these
+cleanup(){
+	kill -n 15 ${PID_SANDBOX}
+	kill -n 15 ${PID_MAC}
+	rm jmxterm-${JMXTERM_VERSION}-uber.jar
+	rm ${BASEDIR}/jmxtermCommands.txt
+	rm -rdf ./scripts/queries/output
+}
+
+check_query(){
+	if [ -z "$1" ];	then
+		echo "No argument supplied to the query function"
+		exit 1
+    else
+    	curl -X POST -u admin:admin -H "Content-Type:application/json" \
+    		-d @./scripts/queries/input/$1.json \
+    		http://localhost:9092/activeviam/pivot/rest/v8/cube/query/mdx \
+    		| jq .cells[].value > ./scripts/queries/output/$1.txt
+    fi
+
+    if [ -z $(diff --strip-trailing-cr --ignore-all-space ./scripts/queries/output/$1.txt ./scripts/queries/ref/$1.txt) ]; then
+    	echo $1"... OK"
+    else
+    	echo "Error when comparing expected query output to query result. Aborting."
+	     # Do the cleanup to make sure we don't leave open processes
+	    cleanup
+	    exit 1
+	fi
+}
+
+# Prepare strings
+MAC_VERSION=3.0.0-SNAPSHOT
+
+AP_ARTIFACTORY_URL=https://activeviam.jfrog.io/artifactory/activepivot-mvn-nightly
+
+AP_VERSION=$1
+AP_REPO_PATH=/com/activeviam/sandbox/sandbox-activepivot/
+
+JMXTERM_VERSION=1.0.4
+JMX_REPO_PATH=https://github.com/jiaqi/jmxterm/releases/download/
+JMX_JAR_PATH=${JMX_REPO_PATH}v${JMXTERM_VERSION}/jmxterm-${JMXTERM_VERSION}-uber.jar
```

Comment:
Maybe define these as env variables in the CI ?

Reply:
Regarding the jmxterm variables Ithink these should stay in this script as the CI should not be aware of the internal technical decisions made within each step 

Regarding the other ones i'll see if i can add them while keeping the script self-enclosed